src/modules/ChannelSettings/context/ChannelSettingsProvider.tsx

```diff
@@ -94,7 +96,38 @@ const ChannelSettingsProvider = ({
       }
 
       try {
-        return await sdkStore.sdk.groupChannel.getChannel(channelUrl);
+        if (channelHandlerId) {
+          if (sdkStore.sdk?.groupChannel?.removeGroupChannelHandler) {
+            logger.info('ChannelSettings: Removing message reciver handler', channelHandlerId);
+            sdkStore.sdk.groupChannel.removeGroupChannelHandler(channelHandlerId);
+          } else if (sdkStore.sdk?.groupChannel) {
+            logger.error('ChannelSettings: Not found the removeGroupChannelHandler');
+          }
+
+          setChannelHandlerId(undefined);
+        }
+
+        const channel = await sdkStore.sdk.groupChannel.getChannel(channelUrl);
+        const channelHandler: GroupChannelHandler = {
+          onUserLeft: (channel, user) => {
+            if (compareIds(channel?.url, channelUrl)) {
+              logger.info('ChannelSettings: onUserLeft', { channel, user });
+              refresh();
+            }
+          },
+          onUserBanned: (channel, user) => {
+            if (compareIds(channel?.url, channelUrl) && channel.isGroupChannel()) {
+              logger.info('ChannelSettings: onUserBanned', { channel, user });
+              refresh();
+            }
+          },
+        };
+
+        const newChannelHandlerId = uuidv4();
+        sdkStore.sdk.groupChannel?.addGroupChannelHandler(newChannelHandlerId, new GroupChannelHandler(channelHandler));
```

Comment:
만약 그렇다면 여기도 옵셔널 체이닝이나 undefined check를 해주면 좋을 것 같습니다
```suggestion
        sdkStore.sdk.groupChannel?.addGroupChannelHandler?.(newChannelHandlerId, new GroupChannelHandler(channelHandler));
```

Reply:
groupChannel이 존재한다면 addGroupChannelHandler는 항상 존재할 것이라고 생각해서 addGroupChannelHandler에 대한 존재성 체크는 하지 않았습니다. groupChannel이 없다면 어차피 기존 코드에서도 undefined로 반환될 거라 불필요하다고 생각했습니다.